misk/src/main/kotlin/misk/flags/memory/InMemoryDynamicFlagStore.kt

```diff
@@ -0,0 +1,42 @@
+package misk.flags.memory
+
+import misk.flags.DynamicFlagStore
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.TimeUnit
+import javax.inject.Singleton
+
+/** In memory implementation of a [DynamicFlagStore], suitable for use in testing */
+@Singleton
+class InMemoryDynamicFlagStore : DynamicFlagStore {
+    private val _booleanFlags = ConcurrentHashMap<String, InMemoryBooleanFlag>()
+    private val _stringFlags = ConcurrentHashMap<String, InMemoryStringFlag>()
+    private val _intFlags = ConcurrentHashMap<String, InMemoryIntFlag>()
+    private val _doubleFlags = ConcurrentHashMap<String, InMemoryDoubleFlag>()
+
+    val booleanFlags: Map<String, InMemoryBooleanFlag> get() = _booleanFlags
```

Comment:
I think the duplication over types here is unnecessary. We wouldn’t support a string and an int flag with the same name; why are they in different maps?

Reply:
(ie. I think you’re working for the type system, it’s not working for you)